src/rgw/rgw_rados.h

```diff
@@ -254,6 +254,14 @@ class RGWObjManifest {
                              as object might have been copied across buckets */
   map<uint64_t, RGWObjManifestRule> rules;
 
+  /* If this object is copied from another versioned object, @src_instance 
+   * is used to store src's version_id. 
+   * 
+   * @copied_obj is used to indicate whether @src_instance is valid.
+   * */
+  string src_instance; 
+  bool copied_obj;
```

Comment:
@yanghonggang ok, I understand the issue now. Instead of calling the field "tail_instance", it should really be use_instance. By default it should match the instance field, unless it's copied and in that case it should match the copied object instance.